### DISCUSSION TITLE
How to add identical keys to search parameters

### DISCUSSION THREAD

**Discussion by IvanHoliak** [TanStack/router#2964] [Q&A] [answered]
Is there such a possibility or can someone tell me how to add several or many identical keys with different values ​​to search parameters?
It might look like this:
`/items?tag=tag1&tag=tag2&tag=tag3 ...etc`

and do navigation:
`
navigate({
  search: (prev) => ({ ...prev, tag: ['tag1', 'tag2'] }),
});
`

and it is possible to do it with zod?

at the moment it was possible to make the URL string looks ugly, just an array of strings in one key
`/items&tag=%5B"win"%2C"triple_kill"%5D`

**Comment by schiller-manuel** [comment, marked as answer]
https://tanstack.com/router/v1/docs/framework/react/guide/custom-search-param-serialization#using-the-query-string-library
